earth_enterprise/src/installer/uninstall_server.sh

```diff
@@ -0,0 +1,352 @@
+  fi
+
+  # Perform backup
+  if [ $BACKUPSERVER == true ]; then
+    backup_server
+  fi
+}
+
+main_uninstall()
+{
+  remove_server_daemon
+  remove_files_from_target
+  change_publish_root_ownership
+  remove_users_groups
+  show_final_success_message
+}
+
+show_help()
+{
+  echo -e "\nUsage:  sudo ./uninstall_server.sh [-h -ndgu -nobk]\n"
+  echo -e "-h \t\tHelp - display this help screen"
+  echo -e "-ndgu \t\tDo Not Delete GEE Users and Group - do not delete the GEE user accounts and group.  Default is to delete both."
+  echo -e "-nobk \t\tNo Backup - do not backup the current GEE Server setup. Default is to backup the setup before uninstalling.\n"
+}
+
+parse_arguments()
+{
+  local retval=0
+
+  while [ $# -gt 0 ]; do
+    case "${1,,}" in
+      -h|-help|--help)
+        show_help
+        retval=1
+        break
+        ;;
+      -nobk)
+        BACKUPSERVER=false
+        ;;
+      -ndgu)
+        DELETE_USERS_GROUPS=false
+        ;;
+      *)
+        echo -e "\nArgument Error: $1 is not a valid argument."
+        show_help
+        retval=1
+        break
+        ;;
+    esac
+
+    if [ $# -gt 0 ]; then
+      shift
+    fi
+  done
+
+  return $retval;
+}
+
+show_intro()
+{
+  echo -e "\nUninstalling $GEES $LONG_VERSION"
+  echo -e "\nThis will remove features installed by the GEE Server installer."
+  echo -e "It will NOT remove files and folders created after the installation."
+}
+
+show_need_root()
+{
+  echo -e "\nYou must have root privileges to uninstall $GEES.\n"
+  echo -e "Log in as the $ROOT_USERNAME user or use the 'sudo' command to run this installer."
+  echo -e "The installer must exit."
+}
+
+show_server_running_message()
+{
+  echo -e "\n$GEES has active running processes."
+  echo -e "You must stop all geserver services before running this installer."
+  echo -e "Please stop the server using the command \"$BININSTALLROOTDIR/geserver stop\".\n"
+}
+
+get_user_names()
+{
+  GEAPACHEUSER_NAME=`cat $BININSTALLROOTDIR/gevars.sh | grep GEAPACHEUSER | cut  -d'=' -f2`
+  GEPGUSER_NAME=`cat $BININSTALLROOTDIR/gevars.sh | grep GEPGUSER | cut  -d'=' -f2`
+  GEFUSIONUSER_NAME=`cat $BININSTALLROOTDIR/gevars.sh | grep GEFUSIONUSER | cut  -d'=' -f2`
+  GEGROUP_NAME=`cat $BININSTALLROOTDIR/gevars.sh | grep GEGROUP | cut  -d'=' -f2`
+
+  # Make sure the users and group exist
+  GEAPACHEUSER_EXISTS=$(getent passwd $GEAPACHEUSER_NAME)
+  GEPGUSER_EXISTS=$(getent passwd $GEPGUSER_NAME)
+  GEFUSIONUSER_EXISTS=$(getent passwd $GEFUSIONUSER_NAME)
+  GEGROUP_EXISTS=$(getent group $GEGROUP_NAME)
+}
+
+check_user_group_delete()
+{
+  local retval=0
+    
+  if [ $DELETE_USERS_GROUPS == true ] && [ $HAS_FUSION == true ]; then
+    echo -e "\nYou cannot delete the GEE server users [$GEAPACHEUSER_NAME, $GEPGUSER_NAME, $GEFUSIONUSER_NAME]\n"
+    echo -e "and group [$GEGROUP_NAME] because $GEEF is installed on this server."
+    echo -e "$GEEF uses these accounts too."        
+    echo -e "\nExiting the uninstaller.\n"
+    retval=1
```

Comment:
I think we need another ticket to look at how the fusion and server uninstallers can work together when both are installed (there is also the problem that some files aren't deleted if fusion is installed).  I can make these changes, though, to make it easier.